src/lib.rs

```diff
@@ -1236,6 +1236,58 @@ pub trait Itertools : Iterator {
     {
         self.sorted_by(cmp)
     }
+
+    /// Return the minimum and maximum element of an iterator.
+    ///
+    /// For the minimum, the first minimal element is returned.  For the maximum,
+    /// the last maximal element wins.  This matches the behavior of the standard
+    /// `Iterator::min()` and `Iterator::max()` methods.
+    ///
+    /// This requires `clone()`-able elements since the returned minimum and
+    /// maximum might actually be the same item.
+    fn minmax(mut self) -> Option<(Self::Item, Self::Item)>
+        where Self: Sized, Self::Item: Ord, Self::Item: Clone
```

Comment:
`f32::min/max` return the non-NaN number if there is one.

I'm not saying this is possible, but I have the ambition :smile: